### ISSUE TITLE
[Accessibility] fd-menu focus issue on click of other interactive elements when dropdown is open

### ISSUE THREAD

**Issue by harshitha-s-kanchan** [SAP/fundamental-ngx#10832]
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
When `fd-menu` dropdown is still open and Input field or interactive div element is clicked, the dropdown should close and the element that is clicked should be focused.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
Angular 15
NGX version -  0.43.11
Issue is also present in latest ngx version

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Issue 1 : Does not focus on input field when input field is clicked when dropdown is open. Focus stays on the fd-menu button.
              Click on Menu and the dropdown opens
              Click on input field when the dropdown is still open
              Observed: Dropdown closes and fd-menu button is focused
              Expected: Dropdown should close and input field should be focused

Issue 2: Does not focus on tabbable interactive div element when div is clicked when dropdown is open. Focus stays on the fd-menu button.
              Click on Menu and the dropdown opens
              Click on interactive div element when the dropdown is still open
              Observed: Dropdown closes and event on the div is triggered, but focus still stays on fd-menu button.
              Expected: Dropdown should close and interactive div element should be focused

#### Please provide relevant source code (if applicable).
The below source code might be relavent for this issue.

https://github.com/SAP/fundamental-ngx/blob/03e181bbc9328d95e197a6a08e5262fa8d5d93fd/libs/core/src/lib/popover/popover-service/popover.service.ts#L144

https://github.com/SAP/fundamental-ngx/blob/03e181bbc9328d95e197a6a08e5262fa8d5d93fd/libs/core/src/lib/popover/popover-service/popover.service.ts#L538-L551
#### Please provide stackblitz example(s).
https://stackblitz.com/edit/vwjges-8dtvgg?file=src%2Fapp%2Fmenu-example.component.html
